bundlewrap/items/pkg.py

```diff
@@ -47,12 +67,15 @@ def pkg_installed(self):
         raise NotImplementedError
 
     def pkg_installed_cached(self):
-        cache = self._pkg_install_cache.setdefault(self.node.name, set())
+        # ensure we don't run pkg_all_installed() concurrently, breaks
+        # some package managers
+        with PKG_INSTALLED_LOCK.lock(self.node.name):
```

Comment:
I decided to add this here, because i believe we have the same issue here.

Reply:
Yes, we do: https://github.com/bundlewrap/bundlewrap/issues/782 👍